tests/components/observers/test_household_survey_observers.py

```diff
@@ -96,6 +96,8 @@ def test_get_observation(
     expected["middle_initial"] = ["J", "M"]
     expected[["guardian_1_address_id", "guardian_2_address_id"]] = np.nan
     expected["survey_date"] = [pd.to_datetime(sim_start_date)] * 2
+    # FIXME: Having dtype with with datime64[s] and [ns] causes pd.testing.assert_frame_equal to fail
+    expected["survey_date"] = expected["survey_date"].astype("datetime64[s]")
```

Comment:
this is a nit, but is there a reason that setting the unit from to_datetime() wouldn't work?

Reply:
The string we are converting to datetime sim_start_date = "2021-01-01" doesn't have the seconds units and I tried a few different things and couldn't get it to work.